indra/newview/llavatarpropertiesprocessor.cpp

```diff
@@ -276,8 +262,13 @@ bool LLAvatarPropertiesProcessor::hasPaymentInfoOnFile(const LLAvatarData* avata
 }
 
 // static
-void LLAvatarPropertiesProcessor::requestAvatarPropertiesCoro(std::string cap_url, LLUUID agent_id)
+void LLAvatarPropertiesProcessor::requestAvatarPropertiesCoro(std::string cap_url, LLUUID agent_id, EAvatarProcessorType type)
 {
+    LLAvatarPropertiesProcessor* self = getInstance();
+
+    // Response received, no longer pending
+    self->removePendingRequest(agent_id, type);
```

Comment:
This is wrong, you are going to free viewer to request again, potentially senconds before request arrives so viewer can rerequest same data multipkle times while it's getting to viewer, at minimumm this needs to be after getAndSuspend, when we know if request failed or not and that it actually happened.